src/main/java/servlets/SearchServlet.java

```diff
@@ -80,10 +80,10 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       return;
     }
 
-    ImmutableList<Receipt> receipts = null;
+    ServerResponse serverResponse = null;
 
     if (checkParameter(request, "isPageLoad")) {
-      receipts = getMatchingReceipts(true);
+      serverResponse = getMatchingReceipts(true);
```

Comment:
I probably should have mentioned this earlier but `getMatchingReceipts(true)` reads as true, this should get matching receipts or `getMatchingReceipts(false)` this should not return matching receipts :). I think ideally I would create separate method names but for now if you just comment the parameter `getMatchingReceipts(/* isPageLoad= */ true)` it would make this a little clearer.

I have a similar comment about `getPage(true)` and `getPage(false)` where I'd probably have methods `getNextPage()` and `getPreviousPage()` instead, but if you could just comment those parameters too that would help.